nxdrive/engine/dao/sqlite.py

```diff
@@ -2305,6 +2338,77 @@ def get_dt_uploads_raw(self, limit: int = 1) -> List[Dict[str, Any]]:
             ).fetchall()
         ]
 
+    def get_session_uploads(self, session_id: int) -> List[Upload]:
+        """
+        Return all active Direct Transfer sessions.
+        Return a simple dict to improve GUI performances.
+        """
+        con = self._get_read_connection()
+        c = con.cursor()
+        return c.execute(
+            "SELECT uid, progress FROM Uploads"
+            f" INNER JOIN States ON Uploads.doc_pair = States.id AND States.session={session_id}"
+        ).fetchall()
+
+    def get_active_sessions_raw(self) -> List[Dict[str, Any]]:
+        """
+        Return all active Direct Transfer sessions.
+        Actives Direct Transfer sessions have the status ONGOING or PAUSED
+        Return a simple dict to improve GUI performances.
+        """
+        con = self._get_read_connection()
+        c = con.cursor()
+        return [
+            {
+                "uid": res.uid,
+                "status": TransferStatus(res.status),
+                "remote_path": res.remote_path,
+                "remote_ref": res.remote_ref,
+                "uploaded": res.uploaded,
+                "total": res.total,
+                "engine": res.engine,
+                "created_at": res.created_at,
+                "completed_at": res.completed_at,
+                "description": res.description,
+            }
+            for res in c.execute(
+                "SELECT * FROM Sessions WHERE"
+                f" status = {TransferStatus.ONGOING.value} OR status = {TransferStatus.PAUSED.value}"
+            ).fetchall()
+        ]
+
+    def get_completed_sessions_raw(
+        self, formatted: bool = False
+    ) -> List[Dict[str, Any]]:
+        """
+        Return all completed Direct Transfer sessions.
+        Completed Direct Transfer sessions have the status DONE or CANCELLED
+        Return a simple dict to improve GUI performances.
+        """
+        con = self._get_read_connection()
+        c = con.cursor()
+        sql_req = (
+            f"SELECT * FROM Sessions WHERE status = {TransferStatus.DONE.value} "
+            f"OR status = {TransferStatus.CANCELLED.value}"
+        )
+        if formatted:
```

Comment:
I would rename `formatted` to `limit` and make it an argument of the method. As done for `get_dt_uploads_raw()`. WDYT?